lib/uplink/encryption.go

```diff
@@ -4,20 +4,184 @@
 package uplink
 
 import (
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/gogo/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/encryption"
+	"storj.io/storj/pkg/macaroon"
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/storj"
 )
 
 const (
 	defaultCipher = storj.EncAESGCM
 )
 
-// EncryptionAccess specifies the encryption details needed to encrypt or
-// decrypt objects.
-type EncryptionAccess struct {
-	// Key is the base encryption key to be used for decrypting objects.
-	Key storj.Key
-	// EncryptedPathPrefix is the (possibly empty) encrypted version of the
-	// path from the top of the storage Bucket to this point. This is
-	// necessary to have in order to derive further encryption keys.
-	EncryptedPathPrefix storj.Path
+// EncryptionCtx represents an encryption context. It holds information about
+// how various buckets and objects should be encrypted and decrypted.
+type EncryptionCtx struct {
```

Comment:
hmm... Is there a reason the name changed? ctx seems more precise but ends up being more convoluted to talk about and write. e.g. `access` vs. `encCtx` for variable names, one seems much easier to communicate.

Reply:
@jtolds ?